frontend/app/helpers/useLoginValidation.ts

```diff
@@ -0,0 +1,73 @@
+import { useState } from "react"
+
+interface UserCredentials {
+  username: string
+  password: string
+}
+
+interface ValidationError {
+  username: string
+  password: string
+}
+
+const useLoginValidation = (): [UserCredentials, ValidationError, (event: React.ChangeEvent<HTMLInputElement>) => void, (event: React.ChangeEvent<HTMLFormElement>) => void] => {
+  const [credentials, setCredentials] = useState<UserCredentials>({
+    username: "",
+    password: ""
+  })
+
+  const [loginFormValidationError, setLoginFormValidationError] = useState<ValidationError>({
+    username: "",
+    password: ""
+  })
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
+    const { name, value } = event.target
+    setCredentials({
+      ...credentials,
+      [name]: value
+    })
+  }
+
+  const handleForm = (event: React.ChangeEvent<HTMLFormElement>): void => {
```

Comment:
I think it was also mentioned in the previous comment to break down `handleForm` function as it is too large.